pkg/ebpf/c/bpf_metadata.h

```diff
@@ -0,0 +1,19 @@
+/**
+ * This header file is included in kconfig.h and ktypes.h, which should make
+ * this metadata available to all object files.
+ */
+
+#ifndef __BPF_METADATA_H__
+#define __BPF_METADATA_H__
+
+#ifndef __CGO__ // Do not include metadata in CGo builds, as it causes duplicate symbols
+#if defined(__x86_64__) || defined(__TARGET_ARCH_x86)
+char __dd_metadata_arch[] __attribute__((section("dd_metadata"), used)) = "<arch:x86_64>";
```

Comment:
What do you think about defining a memory layout here, rather than using strings:
```
struct dd_metadata {
   enum Arch arch;
}
```
This would be extensible and easily marshalled on the go side for validation and other uses.
You also won't have to define the format in two places (c-side and go-side) by exporting the type definition.

Reply:
I thought about something like this, the problem is how to read this without using a dedicated tool, such as the `readelf` case.

Reply:
Hmmm... One idea is to get the section with llvm-objcopy and then marshal the bytes in the test.

Reply:
I think getting `llvm-objdump` in the test images is going to be harder. Also I was thinking that a binary format might be a bit more brittle in terms of versioning, padding... I feel that for having just simple metadata having strings might be easier.

Reply:
If llvm utils are hard to get in the image the I agree, its not worth it.
I think versioning can be easily accomplished with some type aliasing like:
```
struct dd_metadata{
   struct version v;
   ....
};
```